### PR TITLE
Galveston Capital Shock and CGE

### PR DESCRIPTION
**Pull request by Vismayak**
Not sure if this is just taking an unnatural amount of time on my machine...

### CODE REVIEW

File: pyincore/analyses/galvestoncge/galvestoncge.py [hunk 1]
```diff
@@ -0,0 +1,2148 @@
...
+        self.set_result_csv_data("post-disaster-factor-demand", post_disaster_demand,
+                                 name="post-disaster-factor-demand", source="dataframe", index=True)
+        self.set_result_csv_data("gross-income", gross_income, name="gross-income", source="dataframe", index=True)
+        self.set_result_csv_data("household-count", household_count, name="household-count", source="dataframe",
+                                 index=True)
+
+    def get_spec(self):
+        return {
+            'name': 'Galveston-cge',
+            'description': 'CGE model for Salt Lake City.',
```
Comment (longshuicy) [inline pyincore/analyses/galvestoncge/galvestoncge.py:2033]: Typo Salt Lake city should be galveston

File: tests/pyincore/analyses/galvestoncge/test_galvestoncge.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    galveston_cge = GalvestonCGEModel(client)
+
+    galveston_cge.set_parameter("model_iterations", 1)
+    galveston_cge.load_remote_input_dataset("SAM", "63c8398f4b843238326be519")
+    galveston_cge.load_remote_input_dataset("BB", "63a354f54b8432383268cd92")
+    galveston_cge.load_remote_input_dataset("MISCH", "63c838954b843238326be513")
+    galveston_cge.load_remote_input_dataset("EMPLOY", "63c836d94b843238326bd982")
+    galveston_cge.load_remote_input_dataset("JOBCR", "63c837f74b843238326be507")
+    galveston_cge.load_remote_input_dataset("OUTCR", "63c838414b843238326be50d")
+    galveston_cge.load_remote_input_dataset("sector_shocks", "63a358404b8432383268cee0")
```
Comment (longshuicy) [inline tests/pyincore/analyses/galvestoncge/test_galvestoncge.py:20]: Can you post the local files you are using? I can give it a try. Those Ids are still pointing to SLC...

**Review by longshuicy — COMMENTED** [review]
Code structure looks good. Don't forget to add two things
1. docs/source/modules.rst. You can look at the example @navarroc did for another analysis here: https://github.com/IN-CORE/pyincore/pull/279/files#diff-bc1ef7e9cb9f11ba7705ec3a60d320a08679d8e9da78e97613db7b8f8618432c
2. changelog entry

File: tests/pyincore/analyses/galvestoncge/test_galvestoncge.py (hunk 2 above)
Comment (Vismayak) [reply tests/pyincore/analyses/galvestoncge/test_galvestoncge.py:20]: `def run_base_analysis():
    client = IncoreClient(pyglobals.INCORE_API_DEV_URL)
    galveston_cge = GalvestonCGEModel(client)

    sam = Dataset.from_file('/Users/mohanar2/Desktop/NCSA Projects/IN-CORE/Resources/Share/Galveston_2019_CGE_Model/galveston/galveston_data/Galveston_SAM_2019.csv',
                            'incore:CGEsam')
    bb = Dataset.from_file('/Users/mohanar2/Desktop/NCSA Projects/IN-CORE/Resources/Share/Galveston_2019_CGE_Model/galveston/galveston_data/Galveston_CAPCOM_2019.csv',
                           'incore:CGEbb')
    jobcr = Dataset.from_file('/Users/mohanar2/Desktop/NCSA Projects/IN-CORE/Resources/Share/Galveston_2019_CGE_Model/galveston/galveston_data/MiscFile/Galveston_JOBCORE_2019.csv',
                              'incore:CGEjobcr')
    misch = Dataset.from_file('/Users/mohanar2/Desktop/NCSA Projects/IN-CORE/Resources/Share/Galveston_2019_CGE_Model/galveston/galveston_data/MiscFile/Galveston_HH_2019.csv',
                              'incore:CGEmisch')
    employ = Dataset.from_file('/Users/mohanar2/Desktop/NCSA Projects/IN-CORE/Resources/Share/Galveston_2019_CGE_Model/galveston/galveston_data/MiscFile/Galveston_FACTOR_2019.csv',
                               'incore:CGEemploy')
    outcr = Dataset.from_file('/Users/mohanar2/Desktop/NCSA Projects/IN-CORE/Resources/Share/Galveston_2019_CGE_Model/galveston/galveston_data/MiscFile/Galveston_OUTCR_2019.csv',
                              'incore:CGEoutcr')
    sector_shocks = Dataset.from_file('/Users/mohanar2/Desktop/NCSA Projects/IN-CORE/Resources/Share/Galveston_2019_CGE_Model/galveston/galveston_data/sector_shocks.csv',
                                      'incore:capitalShocks')

    galveston_cge.set_parameter("model_iterations", 1)
    # galveston_cge.load_remote_input_dataset("SAM", "63c8398f4b843238326be519")
    galveston_cge.set_input_dataset("SAM", sam)
    # galveston_cge.load_remote_input_dataset("BB", "63a354f54b8432383268cd92")
    galveston_cge.set_input_dataset("BB", bb)
    # galveston_cge.load_remote_input_dataset("MISCH", "63c838954b843238326be513")
    galveston_cge.set_input_dataset("MISCH", misch)
    # galveston_cge.load_remote_input_dataset("EMPLOY", "63c836d94b843238326bd982")
    galveston_cge.set_input_dataset("EMPLOY", employ)
    # galveston_cge.load_remote_input_dataset("JOBCR", "63c837f74b843238326be507")
    galveston_cge.set_input_dataset("JOBCR", jobcr)
    # galveston_cge.load_remote_input_dataset("OUTCR", "63c838414b843238326be50d")
    galveston_cge.set_input_dataset("OUTCR", outcr)
    # galveston_cge.load_remote_input_dataset("sector_shocks", "63a358404b8432383268cee0")
    galveston_cge.set_input_dataset("sector_shocks", sector_shocks)
    galveston_cge.run_analysis()


if __name__ == '__main__':
    run_base_analysis()
`

The link for the data is https://github.com/lutao11/Share/tree/main/Galveston_2019_CGE_Model/galveston/galveston_data

I have attached the testing code I use, so you just need to modify the file path :)

File: tests/pyincore/analyses/galvestoncge/test_galvestoncge.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    galveston_cge.run_analysis()
```
Comment (longshuicy) [inline tests/pyincore/analyses/galvestoncge/test_galvestoncge.py:22]: Make sure you replace those ids with 
```
641e1fccea308903b2b8c51a SAM
641e1fff34810d74880b3985 BB
641e209eea308903b2b8c51b JOBCR
641e20f134810d74880b3986 OUTCR
641e2152ea308903b2b8c51c Employ
641e21a702414a24a5436dfe MISCH
```

It won't work for dev yet but eventually we will migrate the dataset to dev and those will be the id

Comment (longshuicy) [reply tests/pyincore/analyses/galvestoncge/test_galvestoncge.py:22]: Those ids are still Salt lake city ids btw

**Review by longshuicy — CHANGES_REQUESTED** [review]

File: pyincore/analyses/galvestoncge/galvestoncge.py [hunk 4]
```diff
@@ -0,0 +1,2108 @@
...
+        # === end replacing the average shocks on housing services
+
+        iNum = 1
+        # iNum = len(sims.columns)
+        KS00 = KS0.copy()
+
+        for num in range(iNum):
+            KS0.loc[K, I] = KS00.loc[K, I].mul(sims.iloc[:, num])
+            KS0 = KS0.fillna(0.0)
+            run_solver(filename, tmp)
```
Comment (longshuicy) [inline pyincore/analyses/galvestoncge/galvestoncge.py:1974]: Could you catch infeasible and other wrong solver status here? Example you can see what Chris did for SLC
https://github.com/IN-CORE/pyincore/pull/290/files#diff-8047c3f02337ce7d619fd574adc25a744604ea540b0505a25957485c25199ae4R1868

**Review by longshuicy — APPROVED** [review]
Looks good!

**Review by ywkim312 — APPROVED** [review]
ran through until the end without an error

**Review by Rashmil-1999 — APPROVED** [review]

**Pull request merged**